File: archive/kids-with-the-greatest-number-of-candies.js
const kidsWithCandies = function(candies, extraCandies) {
  let max = -Infinity
  candies.forEach(candie => max = Math.max(candie, max))
  let ans = candies.map(candie => {
    if (candie + extraCandies >= max) {
      return true
    } else {
      return false
    }
  })

  return ans
}

console.log(kidsWithCandies([2,3,5,1,3], 3))

File: kids-with-the-greatest-number-of-candies.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
    max := 0; n:=len(candies)
    for i:=0; i < len(candies); i++ {
        if (max < candies[i]) {max = candies[i]}
    }

    arr:=make([]bool, n)
    for i:=0; i < len(candies); i++ {
        if (candies[i] + extraCandies >= max) {
            arr[i] = true
        } else {
            arr[i] = false
        }
    }

    return arr
}